File: 1/c#/TCyllinder.cs
using System;

public class TCyllinder : TCircle{
    public int Height;
	
	//constructors
    public TCyllinder(int _radius, int _height) : base(_radius) {
    	this.Radius = _radius;
		this.Height = _height;
    }
    public TCyllinder() {
    	this.Radius = 0;
		this.Height = 0;        
    }
    public TCyllinder(TCyllinder _cyll0) {
        this.Radius = _cyll0.Radius;
        this.Height = _cyll0.Height;
    }
	//functions
    public void incHeight(int inc){
        int res = this.Height + inc;
        if (res < 0) {
            this.Height = 0;
        }else{
            this.Height = res;
        }
    }
    
    public void mulHeight(double mul){
        this.Height = (int)Math.Round(this.Height*mul);
    }
    
    public int getVolume(){
        return (int)Math.Round(this.Height*Math.PI*this.Radius*this.Radius);
    }
    
	public override int getArea(){
	    return (int)Math.Round(2*Math.PI*this.Radius*(this.Radius+this.Height));
	}
};

File: 1/c++/TCyllinder.cpp
#include "TCircle.cpp"
class TCyllinder : public TCircle {
    public:
		int Height;
	
    	//constructor
    	TCyllinder(void){
    		Radius = 0;
    		Height = 0;
    	}
    	TCyllinder(int _rad, int _height){
    		Radius = _rad;
    		Height = _height;
    	}    
    	TCyllinder(const TCyllinder & _cyll0){
    		Radius = _cyll0.Radius;
    		Height = _cyll0.Height;
    	}    
    	~TCyllinder(){}
    
		//functions
		void incHeight(int inc){
			int res = Height + inc;
			if (res < 0) {
				Height = 0;
			}else{
				Height = res;
			}
		}
		
		void mulHeight(double mul){
			Height = int(round(Height*mul));
		}
		
		int getVolume(){
			return int(round(Height*3.14*Radius*Radius));
		}
		
		int getArea(){
			return int(round(2*3.14*Radius*(Radius+Height)));
		}
};